streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreChangelogReader.java

```diff
@@ -273,8 +273,9 @@ private long processNext(final List<ConsumerRecord<byte[], byte[]>> records,
         long nextPosition = -1;
         int numberRecords = records.size();
         int numberRestored = 0;
+        long offset = 0;
         for (final ConsumerRecord<byte[], byte[]> record : records) {
-            final long offset = record.offset();
+            offset = record.offset();
```

Comment:
I think this is only correct, if `record` is included in the restore. However, if `hasCompleted` returns `true` the record will not be restored and we need to use the offset of the record before?

Reply:
ack